Examples for t.dispatchEvent (#52)
DevExpress/testcafe-examples: examples/dispatch-custom-event/README.md

* the event's constructor
* whether the event is [cancelable](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable)
* whether the event [bubbles](https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles)  

TestCafe invokes the [CustomEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent) to create events, unless the event in question is a [Mouse event](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#mouse-events), a [Keyboard event](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#keyboard-events), an [Input event](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#input-events), a [Focus event](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#focus-events) or a [Pointer event](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#pointer-events).

> Include the `eventConstructor` option to use the [TouchEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/TouchEvent) instead.```

Reviewer: Can you clarify what you mean by "Include"? Are you enabling this option ("Enable the...")? Or do you mean "Add the `eventConstructor` option"?

Author: You pass the option to a method, like so:

```
t.dispatchEvent(..., '...', {eventConstructor: 'EventName})
```

Author: Does this work

```suggestion
To use the [TouchEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/TouchEvent), Pass the `eventConstructor` option to `t.dispatchEvent`.```
```

Reviewer: Thank you. Please apply your suggestion.

> ## TestCafe API methods Used in This Example

Reviewer: ```suggestion
## TestCafe API Methods Used in This Example
```

> # Create Data-Driven Tests

Reviewer 2: `# Fire custom event`

**Test Code**: [index.js](index.js)

**Tested Page**: [index.html](index.html)

This example shows how to fire [events](https://developer.mozilla.org/en-US/docs/Web/API/Event) on the webpage with TestCafe.

When a [touchstart event](https://developer.mozilla.org/en-US/docs/Web/Events/touchstart) is triggered on a target item, the page dislays the event's information:

* event's constructor
* whether the event is [cancelable](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable)
* whether the event [bubbles](https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles)  

> During the test the `t.dispatchEvent` fires a `touchstart` event on the target.

Reviewer 2: Move this sentence under the head.

TestCafe invokes the [CustomEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent) for any event that is neither a [Mouse](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#mouse-events), a [Keyboard](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#keyboard-events), an [Input](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#input-events), a [Focus](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#focus-events) nor a [Pointer](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/dispatchevent.html#pointer-events) event.

> To tell TestCafe to use the [TouchEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/TouchEvent), the `eventConstructor` property is passed to the method's `options`.

Reviewer 3: "To tell..." + passive voice = bad sentence. Better to show than tell.
Could you insert a code snippet that demonstrates this option below?

```suggestion
Include the `eventConstructor` option to use the [TouchEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/TouchEvent) instead.```

> When `touchstart` fires, the `t.expect.eql` method checks the values displayed on the page.

Reviewer 3: why are you saying this and what do you *want* to say? can we just remove this sentence and maybe turn it into a comment? I haven't yet read the source code, and most readers that get to this point will not have read it either. We should either provide some context to make it meaningful or remove it altogether.

```suggestion
```

> ## TestCafe API Used in This Example

Reviewer 3: ```suggestion
## TestCafe API methods Used in This Example
```
